sdk/servicebus/service-bus/src/core/batchingReceiver.ts

```diff
@@ -160,20 +163,29 @@ export class BatchingReceiver extends MessageReceiver {
           // Setting drain must be accompanied by a flow call (aliased to addCredit in this case).
           this._receiver.drain = true;
           this._receiver.addCredit(1);
-        } else {
+
+          const drainTimeout = delay(receiveDrainTimeoutInMs);
+          // Wait for the drain event to be fired, or for the timeout.
+          await Promise.race([drainPromise, drainTimeout]);
```

Comment:
Hey, I like this!

Reply:
Thanks! I think I saw something like this in one of your PRs!